frontend/packages/console-app/src/components/quick-starts/loader/QuickStartsLoader.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { QuickStart } from '../utils/quick-start-types';
+import { getQuickStarts } from '../utils/quick-start-utils';
+import QuickStartPermissionChecker from './QuickStartPermissionChecker';
+
+type QuickStartsLoaderProps = {
+  children: (quickStarts: QuickStart[], loaded: boolean) => React.ReactNode;
+};
+
+const QuickStartsLoader: React.FC<QuickStartsLoaderProps> = ({ children }) => {
+  const [quickStarts, setQuickStarts] = React.useState<QuickStart[]>([]);
+  const [allowedQuickStarts, setAllowedQuickStarts] = React.useState<QuickStart[]>([]);
+  const [loaded, setLoaded] = React.useState<boolean>(false);
```

Comment:
You don't need both of these.
If `allowedQuickStarts` is set, then show, otherwise consider it loading.

Or derive the allowed quick starts in render.

Reply:
Done.